### PR TITLE
Fix missing support for -msse4 alias

### PR DESCRIPTION
**Pull request by giatorta**
See issue [#12746](https://github.com/emscripten-core/emscripten/issues/12746)

### CODE REVIEW

File: tests/test_core.py [hunk 1]
```diff
@@ -5847,6 +5847,19 @@ def test_sse4_2(self):
...
+  def test_sse4(self):
+    src = path_from_root('tests', 'sse', 'test_sse4_2.cpp')
+    self.run_process([shared.CLANG_CXX, src, '-msse4', '-Wno-argument-outside-range', '-o', 'test_sse4_2', '-D_CRT_SECURE_NO_WARNINGS=1'] + clang_native.get_clang_native_args(), stdout=PIPE)
+    native_result = self.run_process('./test_sse4_2', stdout=PIPE).stdout
+
+    orig_args = self.emcc_args
+    self.emcc_args = orig_args + ['-I' + path_from_root('tests', 'sse'), '-msse4', '-Wno-argument-outside-range']
+    self.maybe_closure()
+    self.do_runf(src, native_result)
+
```
Comment (juj) [inline tests/test_core.py:5862]: There already exists a SSE 4.2 test in the suite. This addition duplicates the test with copy-paste, making two almost identical implementations harder to maintain.

Can you remove this and adjust the existing test `test_sse4_2` to test both flags `-msse4` and `-msse4.2`?

File: tools/shared.py [hunk 2]
```diff
@@ -494,6 +494,9 @@ def array_contains_any_of(hay, needles):
     c_opts += ['-D__SSE4_2__=1']
 
   if array_contains_any_of(user_args, SIMD_INTEL_FEATURE_TOWER[6:]):
+    c_opts += ['-D__SSE4_2__=1']
```
Comment (juj) [inline tools/shared.py:497]: The check `if array_contains_any_of(user_args, SIMD_INTEL_FEATURE_TOWER[5:]):` already implies the check `array_contains_any_of(user_args, SIMD_INTEL_FEATURE_TOWER[6:]):`, so this check is redundant.

File: tests/test_core.py (hunk 1 above)
Comment (giatorta) [reply tests/test_core.py:5862]: ok

File: tools/shared.py (hunk 2 above)
Comment (giatorta) [reply tools/shared.py:497]: ok

**Comment by giatorta** [comment]
sorry, I am new: looks like it's stuck?

**Comment by juj** [comment]
Hmm, yes.. can you merge against latest master, that should unblock the CI?

**Comment by kripken** [comment]
CI looks good, but the diff here looks odd. Merging in latest upstream master (from this repo) should fix that.

**Comment by kripken** [comment]
Diff looks good now, thanks @giatorta . Please take another look @juj

**Comment by RReverser** [comment]
I see this got stuck during last review; can we merge it now?

**Comment by sbc100** [comment]
Looks like we have some conflicts again.  I think we can land as soon as they are resolved.

**Review by juj — APPROVED** [review]
Apologies for the delay in review. This looks good, -msse4 indeed should behave as an alias to -msse4.2 in clang.

**Comment by kripken** [comment]
@giatorta This is ready to land aside from the merge conflicts here. Let us know if need help with those.

**Comment by RReverser** [comment]
I believe this is still useful?

**Comment by sbc100** [comment]
Can this be re-based?  (or merged with main?)

**Comment by mr-c** [comment]
https://github.com/emscripten-core/emscripten/commit/4146033c995a505830143fc35a2b69ab6adcd75d was merged, so this PR can be closed

**Pull request closed without merging**